### ISSUE TITLE
Add support for C# target in dotnet core/nuget

### ISSUE THREAD

**Issue by JocaPC** [antlr/antlr4#1515]
I'm trying to use Antlr4 in .Net Core (C# target) but it looks like a dead-end.

I can download jar and generate .cs parser using command line and C# target, however:

- I cannot use Antlr4 .Net runtime from nuget https://www.nuget.org/packages/Antlr4.Runtime/ becuase it is incompatible with the parser generated with this tool. **sharwell** answered in this stack overflow question that this Antlr4 C# generator (which is a fork of his code) is not compatible with his run-time on nuget (although they have the same version 4.5.3!), see http://stackoverflow.com/questions/41138536/antlr-4-5-3-c-sharp-lexer-fails-to-initialize-atn-uuid-error
- I have tried to use runtime from: http://www.antlr.org/download/antlr-csharp-runtime-4.5.3.zip, however, it is a dll, and dotnet core requires dlls to be on nuget. I don't want to crate my own nuget package with your code.
- I have tried to download source code from this repository and put source code; however, it seems that it cannot be compiled in .Net core (~60 errors such as Tuple ConcurentDictionary is ambiguous)

To summarize, although I can create .cs code for parser, I cannot run it, so .Net core is not supported. I will try with alternative target, but please try to make this one working for .Net core.

Thanks,
Jovan

**Comment by ericvergnaud** [comment]
Hi,

release of 4.6 is imminent, and the standard C# runtime will be available from NuGet.
However, it will not support .Net Core yet.

The plan is to make that happen with the next release.

Eric

**Comment by KvanTTT** [comment]
@ericvergnaud could you please also unpload ANTLR parser generator itself for standart runtime? Something like "ANTLR 4 Standart". It's more conveniet for .NET users.

**Comment by ericvergnaud** [comment]
Not sure I can package jars in Nuget...

Envoyé de mon iPhone

**Comment by KvanTTT** [comment]
I suppose there are two ways of such problem resolving:
1. Add a dummy .net library to nuget repository.
2. Add ANTLR 4 Standart Runtime inside ANTLR 4 Standart package.

I prefer the second one.
@sharwell's generator, for example, contains a `Antlr4BuildTasks.net40.dll` build library inside.

**Comment by parrt** [comment]
We uploaded https://www.nuget.org/packages/Antlr4.Runtime.Standard/4.6.0 Can you give it a try?

**Comment by parrt** [comment]
Fixed by https://github.com/antlr/antlr4/pull/1522

**Comment by gortok** [comment]
Can we re-open this? Whenever I try 4.6.0 Antlr.Standard.Runtime or 4.7.0; neither are supported by .NET Core: 
```
Restore completed in 745.23 ms for /src/custom-reports/src/App/StackExchange.DataExplorer/CustomReports.csproj.
/usr/local/share/dotnet/sdk/1.0.3/NuGet.targets(97,5): error : Package Antlr4.Runtime.Standard 4.6.0 is not compatible with netcoreapp1.1 (.NETCoreApp,Version=v1.1). Package Antlr4.Runtime.Standard 4.6.0 supports: net35 (.NETFramework,Version=v3.5) [/src/custom-reports/src/App/CustomReports.Core.sln]
/usr/local/share/dotnet/sdk/1.0.3/NuGet.targets(97,5): error : One or more packages are incompatible with .NETCoreApp,Version=v1.1. [/src/custom-reports/src/App/CustomReports.Core.sln]
```

**Comment by ericvergnaud** [comment]
Hi,

this is getting close, see #1845
